### PR TITLE
1676 Manager pass job_id to processor

### PR DESCRIPTION
**Pull request by francescoingv**
# Overview
Solve issue  #1676 using option 2

# Related Issue / discussion
Issue #1676 

# Dependency policy (RFC2)

- [X] I have ensured that this PR meets [RFC2](https://pygeoapi.io/development/rfc/2) requirements

# Updates to public demo

- [X] I have ensured that breaking changes to the [pygeoapi master demo server](https://github.com/geopython/demo.pygeoapi.io) have been addressed
  - [ ] https://github.com/geopython/demo.pygeoapi.io/blob/master/services/pygeoapi_master/local.config.yml

# Contributions and licensing

(as per https://github.com/geopython/pygeoapi/blob/master/CONTRIBUTING.md#contributions-and-licensing)

- [X] I'd like to contribute [feature X|bugfix Y|docs|something else] to pygeoapi. I confirm that my contributions to pygeoapi will be compatible with the pygeoapi license guidelines at the time of contribution
- [X] I have already previously agreed to the pygeoapi Contributions and Licensing Guidelines

### CODE REVIEW

**Comment by francescoingv** [comment]
@tomkralidis
The reason for the failed check (Build / main (3.10) (pull_request) )
is not related to this PR.

File: pygeoapi/process/base.py [hunk 1]
```diff
@@ -53,6 +53,13 @@ def __init__(self, processor_def: dict, process_metadata: dict):
         self.metadata = process_metadata
         self.supports_outputs = False
 
+    def set_job_id(self, job_id: str) -> None:
+        """
+        set the job_id within the processor
+        To be implemented by derived classes where required.
+        """
```
Comment (tomkralidis) [inline pygeoapi/process/base.py:60]: Can you add a description of `job_id` in the docstring? i.e.

```python
 def set_job_id(self, job_id: str) -> None:
        """
        Sets a job identifier explicitly

        To be implemented by derived classes as required
        """

        :param job_id: job identifier

        :returns: `None`

        pass
```

**Review by tomkralidis — CHANGES_REQUESTED** [review]

File: pygeoapi/process/base.py (hunk 1 above)
Comment (tomkralidis) [reply pygeoapi/process/base.py:66]: @francescoingv thanks for the update.  The return type needs to be added to the docstring.  As well, the `job_id` description needs the `param prefix`, and the text of the `job_id` descripition needs to be aligned.  So:

```python
 def set_job_id(self, job_id: str) -> None:
        """
        Sets a job identifier explicitly

        To be implemented by derived classes as required
        """

        set the job_id within the processor

        To be implemented by derived classes where required.

        :param job_id: the job_id assigned to the request by the Manager.
                    The function shuold be called by the Manager upon assigning
                    the job_id. The job_id is intended to be used by derived
                    classes, e.g. to write temporary files where filenames
                    contains the string job_id.

        :returns: `None`

        pass
```

**Review by tomkralidis — CHANGES_REQUESTED** [review]

File: pygeoapi/process/base.py (hunk 1 above)
Comment (francescoingv) [reply pygeoapi/process/base.py:67]: @tomkralidis Thank you

**Review by tomkralidis — APPROVED** [review]

**Pull request merged**